lib/Comment/EditCommentForm.js

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useRef } from 'react';
+import { arrayOf, bool, func, shape, string } from 'prop-types';
+import { Mention, MentionsInput } from 'react-mentions';
+import Avatar from '../Avatar';
+import AvatarInformation from '../Avatar/AvatarInformation';
+import ShortcutTrigger from '../ShortcutTrigger';
+import Button from '../Button';
+import { MetaDot } from './CommentMeta';
+
+export function EditCommentForm({
+  onSubmit,
+  onRowCountChange,
+  users,
+  autoFocusInput,
+  onCancel,
+  placeholder,
+  initialValue
+}) {
+  const [value, setValue] = useState(initialValue);
+  const [inputHasFocused, setInputHasFocused] = useState(autoFocusInput);
+  const [currentInputHeight, setCurrentInputHeight] = useState('0px');
+  const inputRef = useRef(null);
+
+  const getInputHeight = () => {
+    if (inputRef.current) {
+      return inputRef.current.getBoundingClientRect().height;
+    }
+    return currentInputHeight;
+  };
+
+  const handleChange = event => {
+    const newHeight = getInputHeight();
+    if (newHeight !== currentInputHeight) {
+      onRowCountChange();
+      setCurrentInputHeight(newHeight);
+    }
+    return setValue(event.target.value);
+  };
+
+  const searchForUsers = term => {
+    if (term.trim() !== '' && users.length > 0) {
+      return users.filter(
+        user =>
+          user.name
+            .toLowerCase()
+            .split(' ')
+            .filter(subStr => subStr.lastIndexOf(term, 0) === 0).length > 0 ||
+          user.display.toLowerCase().lastIndexOf(term, 0) === 0
+      );
+    }
+    return users;
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+
+    onSubmit(value);
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <div ref={inputRef}>
+        <MentionsInput
+          className="edit-comment w-full text-sm "
```

Comment:
So `react-mentions`... It seems to take the first class that's passed into `MentionsInput`, and place a class with that name appended with `__input` on the `textarea` that it renders. So to style it I had to write some actual css to target that `textarea`

Reply:
Seems like the only solution if we can't pass tailwind classes to the element - this is bad design on `react-mentions` side

Reply:
The only way I can imagine us not using CSS here is by using the `inputRef` prop that `react-mentions` accepts and passes to it's input element to set the classes with JS.

I don't think this is any cleaner or netter than the CSS solution.